hera_cal/lstbin.py

```diff
@@ -601,39 +673,45 @@ def lst_bin_files(data_files, input_cals=None, dlst=None, verbose=True, ntimes_p
     kwargs['start_jd'] = start_jd
     integration_time = np.median(hd.integration_time)
     assert np.all(np.abs(np.diff(times) - np.median(np.diff(times))) < 1e-6), 'All integrations must be of equal length (BDA not supported).'
-
-    # get baselines from data and form baseline groups
-    bls = sorted(hd.get_antpairs())
-    Nbls = len(bls)
+    # get antpos over all nights
+    for dlist in data_files:
+        hd = io.HERAData(dlist[-1])
+        for a in hd.antpos:
+            if a not in antpos:
+                antpos[a] = hd.antpos[a]
+    # generate a list of dictionaries which contain the nights occupied by each unique baseline
+    # (or unique baseline group if average_redundant_baselines is true)
+    bldicts = gen_bldicts([io.HERAData(dlists[-1]) for dlists in data_files], bl_error_tol=bl_error_tol, include_autos=include_autos,
+                          redundant=average_redundant_baselines)
+    # store all baselines in this list. They need to be spoofed
+    # if there are some times that dont overlap for nights with different baselines.
+    # all_key_baselines = [list(bldict.values())[0] for bldict in blgroups]
+    # iterate over output LST files
     if Nbls_to_load in [None, 'None', 'none']:
-        Nbls_to_load = Nbls
-    Nblgroups = Nbls // Nbls_to_load + 1
-    blgroups = [bls[i * Nbls_to_load:(i + 1) * Nbls_to_load] for i in range(Nblgroups)]
+        Nbls_to_load = len(bldicts)
+    Nblgroups = len(bldicts) // Nbls_to_load + 1
```

Comment:
If `Nbls_to_load` is `None`, then `Nblgroups` is 2... shouldn't it be 1?

Reply:
I've fixed this.